warp10/src/main/java/io/warp10/continuum/egress/EgressExecHandler.java

```diff
@@ -138,7 +139,8 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
     //
     
     WarpScriptStack stack = new MemoryWarpScriptStack(this.storeClient, this.directoryClient);
-
+    stack.setAttribute(WarpScriptStack.ATTRIBUTE_NAME, "[EgressExecHandler " + Thread.currentThread().getName() + "]");
+    
```

Comment:
Why can't they add a prefix to the WarpScript code they push?

Reply:
- It is more simple than prepend lines, shift line numbers or statement number in error messages. VSCode already does macro inlining and error message transformations, studio does not.
- It does not appear in the 'warpscript.script' WarpScript logs.
- More flexible to enforce some extra policy later, for example include the user name in each request, or to add a proxy that automatically add the incoming IP or other metadata in attribute_name if a company want to find who is taking all the shared resources. (128 character is a bit too short for that)

Reply:
You don't need to shift lines as you could very well set the session on the first line without a LF. 
Having an alternate way of setting things just for the /exec endpoint brings redundancy and lack of coherency without added value, therefore this will not be added.

Reply:
Second thoughts, added possibility to set the session via header (but not the name), but disallowed overwriting the session in WSSESSION.

PTAL.